### DISCUSSION TITLE
Future of nvd3 legacy Bar Charts without time axis?

### DISCUSSION THREAD

**Discussion by gbusch** [apache/superset#21333] [Q&A / Help] [answered]
For my dashboard, I need a bar chart that has no time axis, just categories on the x-axis.
I use the "Bar Chart" for this and it works fine. 
However I am confused because these charts, based on nvd3, are labeled as "legacy". The new way seems to be "ECharts". However, I only see bar charts with time on x-axis. 
What is the recommended way?

I am asking because I needed to make some modifications for our customer, namely:
* order bars by some other metric (neither aggregate value nor label)
* use different colors for bars, based on the category

I found some solutions for that and the question is, does it make sense to open a PR for this?

**Comment by cwegener** [comment, marked as answer]
Hi,

In order to use generic X-Axes with the Echarts (v2 charts), you need to enable the Feature Flag for that first.

https://superset.apache.org/docs/installation/configuring-superset#feature-flags

The flag for this feature is called `GENERIC_CHART_AXES`
